src/rgw/rgw_rest_s3.cc

```diff
@@ -527,6 +526,27 @@ int RGWGetObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t bl_ofs,
       decode(rt, iter);
 
       if (rt == rgw::sal::RGWRestoreType::Temporary) {
+        auto expire_iter = attrs.find(RGW_ATTR_RESTORE_EXPIRY_DATE);
+        ceph::real_time expiration_date;
+        rgw::sal::RGWRestoreStatus restore_status;
+        attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
+        if (attr_iter != attrs.end()) {
+          bufferlist bl = attr_iter->second;
+          auto iter = bl.cbegin();
+          decode(restore_status, iter);
+        }
+        if (expire_iter != attrs.end()) {
+          bufferlist bl = expire_iter->second;
+          auto iter = bl.cbegin();
+          decode(expiration_date, iter);
+        }
+        //restore status
+        if (restore_status == rgw::sal::RGWRestoreStatus::CloudRestored) {
+          dump_header_if_nonempty(s, "x-amz-restore", "ongoing-request=\"false\", expiry-date=\""+ dump_time_to_str(expiration_date) +"\"");
+        } else if (restore_status == rgw::sal::RGWRestoreStatus::RestoreAlreadyInProgress) {
```

Comment:
Haven't realized earlier.. this check when moved inside here will not work right.

if restore_status == RestoreInProgress, then restore_type will never be Temporary. And if restore_type is Temporary, for sure restore_status will always be CloudRestored. Need not check again.

So could you please move this restore_status and InProgress check outside this Temporary loop and there is no need to read/check restore_status here  again if the type is Temporary.

Reply:
Done!